hyperspy/model.py

```diff
@@ -271,10 +271,23 @@ def __hash__(self):
         # raise an exception when using windows.connect
         return id(self)
 
-    def __call__(self, non_convolved=False, onlyactive=False, component_list=None, binned=None):
+    def __call__(self, onlyactive=False, component_list=None, binned=None):
         """Evaluate the model numerically. Implementation requested in all sub-classes"""
         raise NotImplementedError
 
+    @property
+    def convolved(self):
+        raise NotImplementedError("This model does not support convolution.")
+    
+    @convolved.setter
+    def convolved(self, value):
+        # This is for compatibility with model saved with HyperSpy < 2.0
```

Comment:
Yes, a model other than an EELSModel will fail on loading. I think that such a situation should be very rare in the wild, since the convolution feature was not documented anywhere other than the sparse documentation in the EELS chapter. Anyway, if it happens, I think that the error will help the user know what's wrong.

I thought on removing the attribute on loading but I think that the current implementations is better because

- While the full feature is now only available in exSpy, parts of it still reside in HyperSpy. Those parts in HyperSpy expect a `Model.convolved` attribute. I think that leaving the attribute is a strong reminder of the situation, that could decrease the risk that in the future somebody reimplements convolution using a different sytax.
- By leaving the attribute, any old script that sets it will raise an informative error, while if it no longer exists, there will be no error.